src/SerializeAST.cc

```diff
@@ -754,10 +774,45 @@ bool IsElementResourceGroup(const Element& element)
     return element.element == Element::CategoryElement && element.category == Element::ResourceGroupCategory;
 }
 
+
+#if _WITH_REFRACT_
+typedef std::vector<const snowcrash::DataStructure*> DataStructures;
+
+void findNamedDataStructures(const snowcrash::Elements& elements, DataStructures& found) {
+    for (snowcrash::Elements::const_iterator i = elements.begin()
+       ; i != elements.end()
+       ; ++i) {
+
+        if (i->element == snowcrash::Element::DataStructureElement) {
+            found.push_back(&(i->content.dataStructure));
+        }
+        else if (!i->content.resource.attributes.empty()) {
+            found.push_back(&i->content.resource.attributes);
```

Comment:
The `content.resource` is an element, right? Then the data structure should be in it's content but you are checking `attributes` here.

Can you add some full API Blueprint tests from drafter.js here so that these kind of things can be checked?